[Kang Yue Hern] iP

yuehernkang opened the pull request:
### Deadline Duck Event Scheduling Chatbot

Download [here](https://github.com/yuehernkang/ip/releases)!

- CLI-Based
- Written in Java

Features:
1. Track **Deadlines**
2. Track **Events**
3. Track **Todos**

Find `Task `Algorithm: 
```   
 /**
 * Searches the list if the description matches the query string
 * @param query query string
 * @return
 */
public List<Task> findTask(String query) {
    List<Task> result = new ArrayList<>();
    Iterator<Task> iterator = this.tasks.iterator();
    while (iterator.hasNext()) {
        Task t = iterator.next();
        System.out.println(t);
        if (t.getDescription().contains(query)) {
            result.add(t);
        }
    }
    return result;
}
```

yuehernkang committed:
Completed Level-1

yuehernkang committed:
Completed Level-2

yuehernkang committed:
Completed Level-3

yuehernkang committed:
Completed Level-4

yuehernkang committed:
Completed Level-5

yuehernkang committed:
Completed Level-6

yuehernkang committed:
A-TextUITesting

yuehernkang committed:
Added File Read/Write capabilities

yuehernkang committed:
Added date time parsing and saving capabilities

Commit:
Merge pull request #1 from yuehernkang/branch-Level-8

Added date time parsing and saving capabilities

yuehernkang committed:
Add UI Class to handle UI Messages

yuehernkang committed:
Add file to duke package

yuehernkang committed:
Add Event adding functionality using OOP

yuehernkang committed:
Add ability to list all tasks with OOP

yuehernkang committed:
Added javadocs comments

Commit:
Merge pull request #2 from yuehernkang/branch-A-JavaDoc

Branch a java doc

yuehernkang committed:
Add DeleteCommand.java

yuehernkang committed:
A-Gradle

Commit:
Merge pull request #3 from yuehernkang/branch-Level-7

Merge pull request #1 from yuehernkang/branch-Level-8

Commit:
Merge pull request #4 from yuehernkang/branch-A-JavaDoc

Add DeleteCommand.java

yuehernkang committed:
Update all files to follow Java coding standard

Commit:
Merge pull request #5 from yuehernkang/branch-A-CodingStandard

Update all files to follow Java coding standard

yuehernkang committed:
Modify Commands to follow OOP for earlier tasks

Commit:
Merge branch 'master' into fix-oop-earlier-tasks

yuehernkang committed:
Merge pull request #6 from yuehernkang/fix-oop-earlier-tasks

Modify Commands to follow OOP for earlier tasks

yuehernkang committed:
Add find functionality

yuehernkang committed:
Merge pull request #7 from yuehernkang/branch-Level-9

Add find functionality

leongdl135 commented on src/main/java/duke/command/CommandParser.java:
```diff
@@ -0,0 +1,23 @@
+public class CommandParser {
+
+    public static String parseCommand(CommandResult commandResult, TaskList taskList, Task t) {
+        switch (commandResult) {
+            case SUCCESSFUL_ADD:
+                return "Got it. I've added this task:\n" + t + "\nNow you have";
+            case SUCCESSFUL_DELETE:
+                break;
+            case SUCCESSFUL_MARK:
+                return "Nice! I've marked this task as done\n" + t;
+            case SUCCESSFUL_UNMARK:
+                break;
```
Should the switch case statements follow a different format? I think you can configure the settings of your ide to suit the coding standards for this module. 😃 
```suggestion
        switch (commandResult) {
        case SUCCESSFUL_ADD:
            return "Got it. I've added this task:\n" + t + "\nNow you have";
        case SUCCESSFUL_DELETE:
            break;
        case SUCCESSFUL_MARK:
            return "Nice! I've marked this task as done\n" + t;
        case SUCCESSFUL_UNMARK:
            break;
```

CJun1039 replied:
Perhaps there should be no indentation for the case clauses?

leongdl135 commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,57 @@
+
+
+/**
+ * Duke class to initialise application
```
Should there be punctuation at the end of each line? I noticed that most of your JavaDocs do not have punctuation. 
```suggestion
 * Duke class to initialise application.
```

leongdl135 commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,82 @@
+
+import java.time.LocalDate;
+
+import duke.command.*;
```
Should wildcard imports be used? I think you can disable wildcard imports in the editor settings in the IDE.

leongdl135 commented on src/main/java/duke/command/CommandResult.java:
```diff
@@ -0,0 +1,9 @@
+package duke.command;
+
+public enum CommandResult {
+    SUCCESSFUL_ADD,
+    SUCCESSFUL_DELETE,
+    SUCCESSFUL_MARK,
+    SUCCESSFUL_UNMARK,
```
Should enum names be in Caps?

leongdl135 commented on src/main/java/duke/ui/Ui.java:
```diff
@@ -0,0 +1,123 @@
+    /**
+     * Display indented dotted line
+     */
+    public void showIndentedDottedLines() {
```
I like the name of this method as it explains to the reader what the methods does at a sufficient level of detail.

leongdl135 reviewed (commented):
Overall, I found your code easy to read and follow. I noted a few minor coding standard violations too. I liked how you separated the code into different packages which made it easier for me to follow. Your code seems to follow the other code quality guidelines taught so far as evident by your good variable/method names. 👍

CJun1039 commented on src/main/java/duke/DateParser.java:
```diff
@@ -0,0 +1,24 @@
+     * Takes the date string and parses it into LocalDate object
+     * @param date
+     * @return LocalDate object for the date string
+     */
```
Should there be an empty line between description and parameter section?

```suggestion
    /**
     * Takes the date string and parses it into LocalDate object
     *
     * @param date
     * @return LocalDate object for the date string
     */
```

CJun1039 commented on src/main/java/duke/command/DeleteCommand.java:
```diff
@@ -0,0 +1,25 @@
+ * Removes Task from the task list
+ */
+public class DeleteCommand extends Command {
+    private final int index;
```
Perhaps constant names should be all uppercase?
```suggestion
    private final int INDEX;
```

CJun1039 commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,78 @@
+    }
+
+    public void deleteTask(int index) {
+        this.tasks.remove(index - 1);
```
Perhaps the `this` keyword should be used only when a field is shadowed by a method or constructor parameter?
```suggestion
        tasks.remove(index - 1);
```

CJun1039 reviewed (commented):
Your code is very readable! Minor styling issues. 👍

AshiqurRah commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,78 @@
+ * Contains the task list and has operations to add/delete tasks in the list
+ */
+public class TaskList {
+    private List<Task> tasks;
```
I like that you named the List<Task> object as tasks, with the plural 's'. Good coding standard

AshiqurRah commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,82 @@
+     * @return
+     */
+    public Command parse(String command) {
+        // Idea from : https://stackoverflow.com/questions/70683058/using-startswith-in-switch-case-in-java
```
I like that you included comments. The comments helped me to understand the code better

AshiqurRah reviewed (commented):
All looks good. Code is readable and easily understandable.

yuehernkang committed:
Add GUI Functionality

yuehernkang committed:
Merge pull request #8 from yuehernkang/branch-Level-10

Add GUI Functionality

yuehernkang committed:
Add more JavaDocs

yuehernkang committed:
Merge pull request #9 from yuehernkang/add-javadocs-comments

Add more JavaDocs

yuehernkang committed:
Add more JavaDocs

yuehernkang committed:
Merge pull request #10 from yuehernkang/add-javadocs-comments

Add more JavaDocs

yuehernkang committed:
Add assertions to some methods

yuehernkang committed:
Merge pull request #11 from yuehernkang/A-Assertions

Add assertions to some methods

yuehernkang committed:
Add mark unmark functionality

yuehernkang committed:
Merge pull request #12 from yuehernkang/Mark-Unmark-Functionality

Add mark unmark functionality

yuehernkang committed:
Follow Code Quality guidelines specified in textbook

yuehernkang committed:
Merge pull request #14 from yuehernkang/branch-A-CodeQuality

Follow Code Quality guidelines specified in textbook

yuehernkang committed:
Add Postpone Task function
- Command for postpone is recognized and adds 1 day to the date
- Streams used for list command

yuehernkang committed:
Enhancement for GUI
- Added background colors for the text based on received or sent message
- Added banner at the top

yuehernkang committed:
Merge pull request #15 from yuehernkang/improvement-to-gui

Enhancement for GUI

yuehernkang committed:
Removed jar files

yuehernkang committed:
Add CI Functionality

yuehernkang committed:
Added Bye, Help and Quack command

yuehernkang committed:
Fix issues related to saving file

yuehernkang committed:
Resolved merge conflicts

yuehernkang committed:
Merge pull request #16 from yuehernkang/B-Snooze

Add Postpone Task function

yuehernkang committed:
Modify parts of the GUI

yuehernkang committed:
Add notes adding functionality

yuehernkang committed:
Add theme to docs

yuehernkang committed:
A-MoreErrorHandling

yuehernkang committed:
A-MoreErrorHandling

yuehernkang committed:
Merge pull request #17 from yuehernkang/error-handling

Error handling

yuehernkang committed:
Create gradle.yml for Java CI workflow

yuehernkang committed:
Modify java dist

yuehernkang committed:
Merge branch 'master' into D-Notes

yuehernkang committed:
Merge pull request #18 from yuehernkang/D-Notes

Add notes adding functionality

yuehernkang committed:
Error handling and bye message

yuehernkang committed:
Improvement to JavaDocs

yuehernkang committed:
Modify JavaDocs comments and add Ui

yuehernkang committed:
Fix find command

yuehernkang committed:
Add additional help messages

yuehernkang committed:
Improvement to Help messages

yuehernkang committed:
Remove comments from Overridden functions

yuehernkang committed:
Update user guide

yuehernkang committed:
Modify duck image

yuehernkang committed:
Modify HelpCommand JavaDocs comment

yuehernkang committed:
Give credit for code reuse and image assets
